packages/ui-kit/src/hooks/use-selected-path.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * Custom hook that manages (for multi paths/languages components) the selected path state and provides a function to update it.
+ * The selected path is stored in local storage and can be updated by dispatching a custom event.
+ * The selected path is also reflected in the URL query string. If a query string activePath parameter is present on page load,
+ * then it overrides the local storage value.
+ * The reason for this double mechanism is that local storage event listeners are not triggered by the same window
+ * that updates the value. See https://developer.mozilla.org/en-US/docs/Web/API/Window/storage_event
+ *
+ * @returns An object containing the selectedPath state and the updateSelectedPath function.
+ */
+import { useState, useEffect } from 'react';
+
+const PATH_LOCAL_STORAGE_KEY = 'selected_path';
+const PATH_CHANGE_EVENT = 'selectedPathChange';
+
+const useSelectedPath = () => {
+  const [selectedPath, setSelectedPath] = useState<string | null>(null);
+
+  useEffect(() => {
+    // Read the initial value from local storage
+    const urlParams = new URLSearchParams(window.location.search);
+    const activePath = urlParams.get('activePath');
+    // if a query string activePath parameter is present, then override the local storage value
+    if (activePath) {
+      localStorage.setItem(PATH_LOCAL_STORAGE_KEY, activePath);
+    }
+    const initialValue = localStorage.getItem(PATH_LOCAL_STORAGE_KEY);
+    setSelectedPath(initialValue);
+
+    // Subscribe to changes in local storage or custom event
+    const handleStorageChange = (event: StorageEvent) => {
+      if (event.key === PATH_LOCAL_STORAGE_KEY) {
+        setSelectedPath(event.newValue?.toLowerCase() || '');
+      }
+    };
+
+    const handleCustomEvent = (event: CustomEvent<string>) => {
+      setSelectedPath(event.detail?.toLowerCase() || '');
+    };
+
+    window.addEventListener('storage', handleStorageChange);
+    window.addEventListener(
+      PATH_CHANGE_EVENT,
+      handleCustomEvent as EventListener
+    );
+
+    // Clean up the event listener
+    return () => {
+      window.removeEventListener('storage', handleStorageChange);
+      window.removeEventListener(
+        PATH_CHANGE_EVENT,
+        handleCustomEvent as EventListener
+      );
+    };
+  }, []);
+
+  const updateSelectedPath = (inputValue: string) => {
+    const value = inputValue.toLowerCase();
+    // Update the value in local storage
+    localStorage.setItem(PATH_LOCAL_STORAGE_KEY, value);
+    setSelectedPath(value);
+
+    // Update the activePath query string parameter
+    const urlParams = new URLSearchParams(window.location.search);
+    urlParams.set('activePath', value);
+    const newLocation =
+      (!window.location.search && !value) || value === ''
+        ? window.location.pathname
+        : `${window.location.pathname}?${urlParams.toString()}`;
+    window.history.replaceState({}, '', newLocation);
```

Comment:
Instead of using history to manage url we can use the URLSearchParams API to set the new location. Can you explain why you want to manage url from history? Its not clear to me

Reply:
Thank you for the suggestion @ColinRosati .

If you have a look to LOC 65 I'm actually using the URLSearchParams api to handle the search parameters update. Using that also guarantees that any other previously exiting parameter in the url is kept while the only update happens to the `activePath` param. 

Regarding the usage of `replaceState`, AFAIK it's needed as the URLSearchParams API concern is just to manipulate the search parameters but not updating the actual browser url.

Finally have a look to the refactored function, it should guarantee that if any parameter !== 'activePath' exists in the url, remains untouched

Reply:
I also added some "more meaningful" names for the `replaceState` parameters, however [the API signature itself it's pretty ugly](https://developer.mozilla.org/en-US/docs/Web/API/History/replaceState#parameters)